### PR TITLE
Explore: Fix Cannot read property 'emit' of null error

### LINKED ISSUE #31585: Explore: TypeError: Cannot read property 'emit' of null
**What happened**:
[From Sentry (7 times)](https://docs.google.com/spreadsheets/d/1xpUL9WLO01fX-FxOl-9sb1omobiQeIq3CYDFV6FHd6g/edit#gid=1280620028):
`TypeError: Cannot read property 'emit' of null`
`at emit (core|webpack:///./public/app/features/explore/state/query.ts:666:22)`

### PR DESCRIPTION
**Pull request by ivanahuckova**
**What this PR does / why we need it**:
From [sentry](https://docs.google.com/spreadsheets/d/1xpUL9WLO01fX-FxOl-9sb1omobiQeIq3CYDFV6FHd6g/edit#gid=1280620028) we learned about this error
`TypeError: Cannot read property 'emit' of null
at emit (core|webpack:///./public/app/features/explore/state/query.ts:666:22)`

Several SQL datasources are listening on those events and therefore we shouldn't get rid of this. The proposed fix is to check if data source have angular editor and only in that case use `  state.eventBridge.emit(PanelEvents.dataError, error)`.  The same logic is used just a couple lines below:
```
  // Send legacy data to Angular editors
  if (state.datasourceInstance?.components?.QueryCtrl) {
    const legacy = series.map((v) => toLegacyResponseData(v));
    state.eventBridge.emit(PanelEvents.dataReceived, legacy);
  }
```

**Which issue(s) this PR fixes**:

Fixes https://github.com/grafana/grafana/issues/31585

### CODE REVIEW

**Review by ifrost — APPROVED** [review]
👍  Do we need `console.log`? If yes, should it be added to `state.eventBridge.emit(...)` a few lines below, too?

File: public/app/features/explore/state/query.ts [hunk 1]
```diff
@@ -660,8 +661,11 @@ export const processQueryResponse = (
       return state;
     }
 
-    // For Angular editors
-    state.eventBridge.emit(PanelEvents.dataError, error);
+    // Send error to Angular editors
+    if (state.datasourceInstance?.components?.QueryCtrl) {
+      console.log('error angular', error);
```
Comment (ivanahuckova) [inline public/app/features/explore/state/query.ts:666]:
```suggestion
```

File: public/app/features/explore/state/query.ts [hunk 2]
```diff
@@ -650,6 +650,7 @@ export const processQueryResponse = (
   } = response;
 
   if (error) {
+    console.log('error', error);
```
Comment (ivanahuckova) [inline public/app/features/explore/state/query.ts:653]:
```suggestion
```

**Comment by ivanahuckova** [comment]
>Do we need console.log? If yes, should it be added to state.eventBridge.emit(...) a few lines below, too?

Forgotten `console.logs` 🙈 . I have removed them.

**Pull request merged**